src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -655,6 +654,8 @@ void GcsActorManager::ReconstructActor(const ActorID &actor_id, bool need_resche
               ACTOR_CHANNEL, actor_id.Hex(),
               mutable_actor_table_data->SerializeAsString(), nullptr));
         }));
+    // Empty address so that we can avoid scheduler schedules at the same address.
+    actor->UpdateAddress(rpc::Address());
```

Comment:
I think we need to update the address before writing to the storage. Otherwise, after the GCS server restarts, it will mistakenly think that the actor has succeeded in the lease worker.

Reply:
I see. So are you suggesting 
```
copy actor table data
update the mutable data
publish the copied data
``` ?

Does it make sense?

Reply:
@ffbin I copied the actor table data before updating some other states. Can you review one more time?